### PR TITLE
Frontend Kiosk Auth UI

### LINKED ISSUE #16: Kiosk UI: Auth UI
UI components for authentication and associated state management. For now, this won't have backend connection (https://github.com/orgs/cs350-team4/projects/1/views/1?pane=issue&itemId=28173255).

- [x] Manual login
- [x] QR login
- [x] Authentication state management

### PR DESCRIPTION
**Pull request by Non-J**
Closes #16 

The goal for this PR is to be lay a groundwork for future frontend components, which can be developed by copying existing components in this PR. The HTTP API is currently using a mocked version (only on storybook) and can be changed/corrected once the corresponding part of the backend is done.

This PR includes:

- UI for logging in via QR code
- UI for logging in via password
- Authentication state system (frontend only)
- Mocked HTTP API

### CODE REVIEW

**Comment by m-spitfire** [comment]
I tested both login via password and QR code. They work! Thanks. I saw that you actually mocked the backend. Wow.

File: frontend/src/data/auth.ts [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+    data: pkQueryData,
+    isLoading: pkQueryLoading,
+  } = useQuery({
+    refetchOnWindowFocus: false,
+    cacheTime: PUBLIC_KEY_CACHE_TIME,
+    queryKey: ["repoData"],
+    queryFn: async () => {
+      const res = await fetch(API_ROOT + "/auth/publicKey");
+      if (res.ok) {
+        return jose.importSPKI(await res.text(), "ES256");
```
Comment (m-spitfire) [inline frontend/src/data/auth.ts:124]: Response isn't publicKey as plain text, but a json object with public key as `publicKey` key(lol it sounds funny)

Comment (Non-J) [reply frontend/src/data/auth.ts]: done

**Comment by m-spitfire** [comment]
@Non-J what's there left for merging?

**Pull request merged**